### PR TITLE
Add UI to remove and add favourite locations in search

### LINKED ISSUE #272: Add ui to remove and add favourite locations in search
Needs #91 #253 

- [x] Implement ui to add and remove favorites in search results
- [x] Display favorites when no search term is entered

### PR DESCRIPTION
**Pull request by Paulomart**
https://github.com/Oeffis/Oeffis/assets/4148404/0ff54ea9-315c-4c0b-a9f0-315412f5bab6

### Description

Closes #272 

## Checklist

- [x] My code follows the style guidelines
- [x] I have documented my code
- [x] My changes generate no new warnings
- [x] I have linked the Issue
- [x] The Pull Request covers all the Checkpoints from the tagged Issue
- [x] I have noted the hours of work i have put in the issue

## I have edited following Parts of the project

- [x] Frontend
- [ ] Backend
- [ ] Github
- [ ] Root

## If tests are required did you make them?

- [ ] No tests required
- [ ] Yes
- [x] No

### CODE REVIEW

**Review by timvahlbrock — APPROVED** [review]
Nice work. Especially that you already included showing the favourites when no search query is put in.

File: .vscode/settings.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
   ],
   "git.branchProtectionPrompt": "alwaysCommitToNewBranch",
   "git.pullBeforeCheckout": true,
-  "git.mergeEditor": true,
```
[the remark this answers is not in the document]
Comment (timvahlbrock) [reply .vscode/settings.json:12]: sorry but this need to be fixed before.

**Review by timvahlbrock — CHANGES_REQUESTED** [review]

**Comment by Paulomart** [comment]
I think its better to merge #281 first and then come back to this pr.

File: frontend/src/components/LocationSearch/LocationIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  "stop": { icon: bus, text: "Stop" },
+  "street": { icon: location, text: "Street" },
+  "suburb": { icon: business, text: "Suburb" },
+
+  // not present in vrr spec
+  "singlehouse": { icon: home, text: "Single House" }
+};
+
+export type LocationIconProps = {
+  type: string;
```
Comment (timvahlbrock) [inline frontend/src/components/LocationSearch/LocationIcon.tsx:24]: we could get more precise by using `keyof ICON_MAP` here, but I don't know if react bindings are that precise

**Review by timvahlbrock — CHANGES_REQUESTED** [review]

**Review by timvahlbrock — APPROVED** [review]

**Pull request merged**